pkg/gardenadm/staticpod/translator.go

```diff
@@ -66,27 +65,28 @@ func translateVolumes(ctx context.Context, c client.Client, pod *corev1.Pod) ([]
 	var (
 		files               []extensionsv1alpha1.File
 		addFileWithHostPath = func(hostPath, fileName, content string, desiredItems []corev1.KeyToPath) {
-			path := filepath.Join(hostPath, fileName)
-			if len(desiredItems) == 0 || slices.ContainsFunc(desiredItems, func(item corev1.KeyToPath) bool {
-				path = filepath.Join(hostPath, item.Path)
+			if len(desiredItems) == 0 {
+				files = append(files, extensionsv1alpha1.File{
+					Path:        filepath.Join(hostPath, fileName),
+					Permissions: ptr.To[uint32](0600),
+					Content:     extensionsv1alpha1.FileContent{Inline: &extensionsv1alpha1.FileContentInline{Data: content}},
+				})
```

Comment:
This part is duplicated with the case below. The only difference is the file name.
Can deduplicate this by overwriting the `filename` if the key is in the `desiredItems` slice and early exit if not?